### PR TITLE
Audible emotes

### PR DESCRIPTION
**Pull request by Macoron**
## About the PR 
Adds new framework for emotes and their audio. I tried to make it as universal as possible. It should help to write systems that can handle emotes using custom logic (override sound, flip sprites, randomize stuff, etc).

All game logic should be exactly same. I decided to leave emotes selection UI outside this PR scope.

<details>
  <summary>Tech details</summary>

Emotes can be defined using new `EmotePrototype`:
```yaml
- type: emote
  id: Scream
  category: Vocal
  chatMessages: [screams, yells]
  chatTriggers:
    - scream
    - screams
    - screaming
    - screamed
```

This prototype doesn't use fluent, because different languages has different forms of verbs (depending on time, gender, etc). So some forks will have to translate this yaml directly.

When player typed emote or some code forced entity to emote, chat system raises `EmoteEvent`. Other systmes listen to this event and can do custom stuff. For example here is how `BodyeEmotesSystem` checks that entity has hands to do clapping sound:
```csharp
private void OnEmote(EntityUid uid, BodyEmotesComponent component, ref EmoteEvent args)
{
    if (args.Handled)
        return;

    var cat = args.Emote.Category;
    if (cat.HasFlag(EmoteCategory.Hands))
    {
        args.Handled = TryEmoteHands(uid, args.Emote, component);
    }
}

private bool TryEmoteHands(EntityUid uid, EmotePrototype emote, BodyEmotesComponent component)
{
    // check that user actually has hands to do emote sound
    if (!TryComp(uid, out HandsComponent? hands) || hands.Count <= 0)
        return false;

    return _chat.TryPlayEmoteSound(uid, component.Sounds, emote);
}
``` 

In most cases you would use `EmoteSoundsPrototype` which is collection that contains emote - sound mapping:
```yaml
- type: emoteSounds
  id: MaleHuman
  params:
    variation: 0.125
  sounds:
    Scream:
      collection: MaleScreams
    Laugh:
      collection: MaleLaugh
```

You can force entity to emote using new chat system API. For example:
```csharp
if (ShowInChat)
    chatSys.TryEmoteWithChat(args.SolutionEntity, EmoteId, hideGlobalGhostChat: true);
else
    chatSys.TryEmoteWithoutChat(args.SolutionEntity, EmoteId);
```
</details>

**Screenshots**

**Changelog**

:cl:
- add: Added new emotes with sound! Simply type into emote chat: scream, laugh, clap or snap.

### CODE REVIEW

**Comment by Macoron** [comment]
I decided to leave UI emotes selection out of scope for this PR. Need to more ideaguy where to place this UI and how it should work.

Code stuff is ready for review.

**Comment by mirrorcult** [comment]
a good UI for this kind of thing (although high effort) would IMO be like how stardew valley emotes work
image
image

File: Resources/Prototypes/Voice/speech_empote_sounds.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# species
```
Comment (mirrorcult) [inline Resources/Prototypes/Voice/speech_empote_sounds.yml:1]: filename is misspelled

**Review by mirrorcult — APPROVED** [review]
really sorry for lack of review, looks fine

**Comment by Visne** [comment]
I just fixed merge conflicts and fixed the filename type. Summary of the merge:

- Changes from #13429 were removed, because this PR already handles that itself.
- Moved SetSex() to SharedHumanoidAppearanceSystem, because it is now networked (#13009)
- Other conflicts were trivial.

**Pull request merged**